src/ngraph_nnvm_utils.h

```diff
@@ -77,22 +77,56 @@ inline TensorVector make_ngraph_placeholders(
   return out;
 }
 
+inline std::shared_ptr<ngraph::runtime::Tensor> get_tensor(
+    std::shared_ptr<Graph>& graph, bool is_fwd, const mxnet::NDArray& ndarray,
+    bool is_boolean, bool is_scalar) {
+  auto create_tensor = [&]() {
+    std::shared_ptr<ngraph::runtime::Tensor> tensor;
+    auto backend = graph->get_backend();
+    void* dptr = ndarray.storage_handle().dptr;
+    check(backend != nullptr);
+    check(dptr != nullptr);
+    ngraph::Shape shape{};
+    if (!is_scalar) {
+      shape = TShape_to_NShape(ndarray.shape());
+    }
+    if (is_boolean) {
+      tensor = backend->create_tensor(ngraph::element::boolean, shape, dptr);
+
+    } else {
+      tensor = backend->create_tensor(getType(ndarray.dtype()), shape, dptr);
+    }
+    return tensor;
+  };
+  auto index =
+      is_fwd ? graph->tensor_fwd_index_cur_++ : graph->tensor_bwd_index_cur_++;
```

Comment:
Is `tensor_*_index_cur_` always increasing as `get_tensor()` called multiple times?

Reply:
NVM, I see you reset it to 0 for each fwd and bwd call above.

Reply:
we reset at beginning of fwd or bwd compute, as the number of ndarrays (inputs or outputs) are fixed at bind.

Reply:
The logic looks good. But I wonder if getting rid of the class data `tensor_*_index_cur_` and use a parameter to pass the index to `get_tensor()` would be easier to understand and avoids using external variables when keeping those states are not necessary.

Reply:
since tensor_*_index_cur_ is tied to tensors, it seem to help to keep both states (tensors and their current index) together in the class.

Reply:
`I see tensor_*_index_cur_ are only used in get_tensor()`

we use/modify it in separate locations/functions (eg: it is reset in compute_fwd/bwd funcs, and updated in get_tensor)